### DISCUSSION TITLE
i can't move cursor by press up/down and left/right arrow

### DISCUSSION THREAD

**Discussion by zhajio1988** [Textualize/textual#2475] [Q&A] [answered]
```
from textual.app import App, ComposeResult
from textual.widgets import DataTable

ROWS = [
    ("lane", "swimmer", "country", "time"),
    (4, "Joseph Schooling", "Singapore", 50.39),
    (2, "Michael Phelps", "United States", 51.14),
    (5, "Chad le Clos", "South Africa", 51.14),
    (6, "László Cseh", "Hungary", 51.14),
    (3, "Li Zhuhao", "China", 51.26),
    (8, "Mehdy Metella", "France", 51.58),
    (7, "Tom Shields", "United States", 51.73),
    (1, "Aleksandr Sadovnikov", "Russia", 51.84),
    (10, "Darren Burns", "Scotland", 51.84),
]


class TableApp(App):
    def compose(self) -> ComposeResult:
        yield DataTable()

    def on_mount(self) -> None:
        table = self.query_one(DataTable)
        table.add_columns(*ROWS[0])
        table.add_rows(ROWS[1:])


app = TableApp()
if __name__ == "__main__":
    app.run()
```
Above code is data_table example, i find i can't move cursor by press up/down and left/right arrow, i don't know why? Could you help solve it?

**Comment by davep** [comment, marked as answer]
Are you sure the `DataTable` had focus? By default, when a Textual application starts up, no widget will have focus, and so in this case the `DataTable` won't receive keyboard input. If you were to press <kbd>Tab</kbd> first and then use the cursor keys you should find that it works (testing here that's exactly what I see with your code above).

To make it so you don't have to press <kbd>Tab</kbd> first you could add the following to the end of your `on_mount` method:

```python
table.focus()
```

**Reply by zhajio1988** [reply]
Cool，it works. Due to time constraints, I quickly looked for examples at https://textual.textualize.io/widgets/data_table to run down, I did not notice the need to press the tab key to work, thank you!

**Reply by davep** [reply]
Yup, it's not always obvious. As I say, right now, it's the startup state for Textual apps, and often a reason why it's a good idea to have some sort of styling for parts of your application that show where focus us.

There's a very good chance [we'll change this soon so it's less confusing](https://github.com/Textualize/textual/issues/2457).
